docs/concepts/select.md

```diff
@@ -353,6 +353,69 @@ export class CityService {
 
 now we can use this `zooThemeParks` selector anywhere in our application.
 
+### Inheriting selectors  
+
+When we have states that share similar structure, we can extract the shared selectors into a base class which we can later extend from. If we have an `entities` field on multiple states, we can create a base class containing a dynamic `@Selector()` for that field, and extend from it on the `@State` classes like this.
+
+```TS
+export class EntitiesState {
```

Comment:
It doesnt allow me to use generics on static members

![image](https://user-images.githubusercontent.com/6530160/61870653-005e3b80-aeb5-11e9-8d1a-3633dd455ad2.png)

Reply:
Yeah it would require to declare `entities` not as property, but as a method:
```ts
static entities<T>() { ... }
```

So we could use it like:
```ts
@Select(State.entities) --> @Select(State.entities<Model>())
```
@splincode @markwhitfeld cc

Reply:
Good point 

```ts
static entities<T>() { ... }
```

Reply:
I prefer typing it on the Component, since the @Select doesn't provide the Observable typing by default. But if you guys want i can update it.

Reply:
what about this 

```
export interface EntitiesStateModel<T> {
  entities: T[]
}

export class EntitiesState {

  static entities<T>() {
    return createSelector([this], (state: EntitiesStateModel<T>) => {
      return state.entities as T[];
    });
  }

}

export interface User {
  name: string
}

export interface UsersStateModel extends EntitiesStateModel<User> {

}
```